doc/source/notebooks/gpflow_2_upgrade/gpflow2_upgrade_guide.ipynb

```diff
@@ -0,0 +1,299 @@
+    "<img src=\"param.png\">\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model Class\n",
+    "\n",
+    "The `Model` class has been removed. A suitable replacement, for those models that do not wish to inherit from `GPModel`, may be `BayesianModel`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Periodic Base Kernel\n",
+    "\n",
+    "The base kernel for the `Periodic` kernel must now be specified explicitly. Previously the default was  `SquaredExponential`, so to maintain the same behaviour as before this must be passed-in to the `Periodic` kernel’s initialiser. \n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"periodic_1.png\">\n",
+    "<img src=\"periodic_2.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Predict Full Covariance\n",
+    "\n",
+    "The `predict_f_full_cov` method has been removed from `GPModel`. Instead, pass `full_cov=True` to the `predict_f` method.\n",
+    "\n",
+    "For example:\n",
+    "\n",
+    "<img src=\"full_cov.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Data Types\n",
+    "\n",
+    "In some cases TensorFlow will try to figure out an appropriate data type for certain variables. If Python floats have been used, TensorFlow may default these variables to `float32`, which can cause incompatibilities with GPflow, which defaults to using `float64`.\n",
+    "\n",
+    "To resolve this you can use `tf.constant` instead of a Python float, and explicitly specify the data type. For example:\n",
+    "\n",
+    "<img src=\"constant.png\">"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Float and Int Types\n",
+    "\n",
+    "In GPflow 2 `gpflow.settings.float_type` has changed to `gpflow.default_float()` and `gpflow.settings.int_type` has changed to `gpflow.default_int()`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Transforms\n",
+    "\n",
+    "These have been removed in favour of the tools in `tensorflow_probability.bijectors`. See for example [this Stackoverflow post](https://stackoverflow.com/q/58903446/5986907).\n",
+    "\n",
+    "GPflow 2 still provides the `gpflow.utilities.triangular` alias for `tfp.bijectors.FillTriangular`, and `gpflow.utilities.positive` which is configurable to be either softplus or exp, with an optional shift to ensure a lower bound that is larger than zero."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Name Scoping\n",
+    "\n",
+    "The `name_scope` decorator does not exist in GPflow 2 anymore. Use TensorFlow’s [name_scope](https://www.tensorflow.org/api_docs/python/tf/name_scope?version=stable) context manager instead."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Model Persistence\n",
```

Comment:
Here, it is also important to say about checkpointing. This is one of the most important ingredients for research. You can say something that there are two ways of saving a model: tf.saved_model and tf.train.Checkpoint and give a links to TensorFlow docs.

Reply:
> Here, it is also important to say about checkpointing. This is one of the most important ingredients for research. You can say something that there are two ways of saving a model: tf.saved_model and tf.train.Checkpoint and give a links to TensorFlow docs.

I've mentioned checkpointing (and that it currently works fine, which I'm assuming is the case) and linked to the TensorFlow docs.